lib/iris/analysis/__init__.py

```diff
@@ -240,7 +240,12 @@ def coord_comparison(*cubes):
         print('All equal coordinates: ', result['equal'])
 
     """
-    all_coords = [cube.coords() for cube in cubes]
+    if object_get is None:
```

Comment:
This is a public function, so you really need to extend the docstring to cover the changes.
It might also be possible to rename some things here, since e.g. "coords" is no longer always what it says.

Another way is to create an enhanced, private inner routine + leave the public one as a thin wrapper for coords alone.

Reply:
I've opened #3558 to address this.